src/provider/dependency_resolution.rs

```diff
@@ -66,13 +125,174 @@ impl Project {
+                let decompiled_locations = decompiled_locations.lock().unwrap();
+                let decompiled_files = &(*decompiled_locations);
+                for decompiled_file in decompiled_files {
+                    debug!("loading file {:?} into database", &decompiled_file);
+                    let stats = load_database(decompiled_file, db.to_path_buf());
+                    debug!("loaded file: {:?} stats: {:?}", &decompiled_file, stats);
+                }
+            }
+        }
+        Ok(())
+    }
+}
+
+impl Dependencies {
+    pub async fn decompile(&self, reference_assmblies: PathBuf) -> Result<(), Error> {
+        // TODO: make location of ilspycmd decompilation
+        let dep_package_dir = self.location.to_owned();
+        if !dep_package_dir.is_dir() || !dep_package_dir.exists() {
+            return Err(anyhow!("invalid package path: {:?}", dep_package_dir));
+        }
+        let mut entries = fs::read_dir(dep_package_dir).await?;
+        let mut paket_cache_file: Option<PathBuf> = None;
+        while let Some(entry) = entries.next_entry().await? {
+            // Find the paket_installmodel.cache file to read
+            // and find the .dll's
+            if entry.file_name().to_string_lossy() == "paket-installmodel.cache" {
+                paket_cache_file = Some(entry.path());
+                break;
+            }
+        }
+        let to_decompile_locations = match paket_cache_file {
+            Some(cache_file) => {
+                // read_cache_file to get the path to the last found dll
+                // this is an aproximation of what we want and eventually
+                // we will need to understand the packet.dependencies file
+                self.read_packet_cache_file(cache_file).await?
+            }
+            None => {
+                debug!("did not find a dll for dep: {:?}", self);
+                return Ok(());
+            }
+        };
+        let mut decompiled_files: HashSet<PathBuf> = HashSet::new();
+        for file_to_decompile in to_decompile_locations {
+            let decompiled_file = self
+                .decompile_file(&reference_assmblies, file_to_decompile)
+                .await?;
+            decompiled_files.insert(decompiled_file);
+        }
+
+        let mut guard = self.decompiled_location.lock().unwrap();
+        *guard = decompiled_files;
+        drop(guard);
+
+        Ok(())
+    }
+
+    async fn read_packet_cache_file(&self, file: PathBuf) -> Result<Vec<PathBuf>, Error> {
+        info!("Reading packet cache file: {:?}", file);
+        let file = File::open(file).await;
+        if let Err(e) = file {
+            error!("unable to find error: {:?}", e);
+            return Err(anyhow!(e));
+        }
+        let reader = BufReader::new(file.ok().unwrap());
+        let mut lines = reader.lines();
+        let mut dlls: Vec<String> = vec![];
+        let top_of_version = "D: /lib/net45".to_string();
```

Comment:
This "version" needs to be either obtained from the project or specified as an init parameter. I'm thinking about this as a follow-up for now.